functions/cloud_functions/user.js

```diff
@@ -38,7 +38,7 @@ exports.userCreated = functions.auth.user().onCreate(async (user) => {
         ROI : 0,
         assignedTrader : "",
         isApproved : false,
-        referralCode : "",
+        referralCode : user.referralCode,
```

Comment:
There is nothing like `user.referralCode`. This is a register function.... check the available `properties` and `methods` of user... there are linked to firebase ... not the form